src/common.rs

```diff
@@ -57,5 +59,5 @@ pub struct Location {
 /// Location provider
 pub trait LocationProvider {
   /// Location fetcher
-  fn location(&self) -> Result<Location, Error>;
+  fn location(&self, config: &Option<Configuration>) -> Result<Location, Error>;
```

Comment:
Suggest converting this to `Option<&T>`, still getting my rust footing but a reference to an option seems out of place. 
If this change is made it would require the caller to do `self.config.as_ref()`
```suggestion
  fn location(&self, config: Option<&Configuration>) -> Result<Location, Error>;
```

Reply:
Yeah, my recollection is that rust is now smart enough to know that `None` vs. `Some(&foo)` is the same as `foo* == nullptr vs. foo* != nullptr`.